primitives/src/vid/namespace.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2023 Espresso Systems (espressosys.com)
+// This file is part of the Jellyfish library.
+
+// You should have received a copy of the MIT License
+// along with the Jellyfish library. If not, see <https://mit-license.org/>.
+
+//! Trait for namespace functionality in Verifiable Information Retrieval (VID).
+
+use super::{VidResult, VidScheme};
+
+/// Namespace functionality for [`VidScheme`].
+pub trait Namespacer: VidScheme {
+    /// data proof
+    type DataProof;
+
+    /// Compute a proof for `payload` for data index range `start..start+len-1`.
+    ///
+    /// See TODO in `namespace_verify` on `payload`.
+    fn data_proof(
+        &self,
+        payload: &Self::Payload,
+        start: usize,
+        len: usize,
+    ) -> VidResult<Self::DataProof>;
```

Comment:
The prover (e.g. query server) also needs some way to get the actual data. So I think this should be something like `VidResult<(Vec<u8>, Self::DataProof)>`, or `Self::DataProof` should have some trait bound that allows me to extract the data from the proof

Reply:
My understanding is that the query server will already have `payload` in full. Thus, it already has the actual data---it could grab its own subslice of `payload`.

But this raises a related issue: Due to bytes-to-field encoding, the user needs a few bytes outside the tx/namespace in order to verify the proof. Where should these bytes go? I guess it makes sense to put those few bytes inside the `DataProof`.

But it's not clear to me that the actual data should also go into the `DataProof`.

Reply:
Yeah, putting the extra bytes in `DataProof` and having the payload data separate seems fine. But the query server still needs some interface for actually reading bytes from the full payload. I don't see that here since `Self::Payload` doesn't have a trait bound that would allow me to read bytes out of it

Reply:
Good point. I'll add something for that.

Reply:
As discussed offline I'll remove the `Payload` assoc type and just enforce `&[u8]` payload. It hurts my soul to be so opinionated in the API, but we discussed that downstream users are assumed to bear the cost of serializing to a 1D byte slice, and this assumption makes our life much easier.